platforms/jvm/scala/src/integTest/groovy/org/gradle/scala/compile/UpToDateScalaCompileIntegrationTest.groovy

```diff
@@ -34,6 +35,7 @@ class UpToDateScalaCompileIntegrationTest extends AbstractIntegrationSpec implem
         file('src/main/scala/Person.scala') << "class Person(name: String)"
     }
 
+    @Requires(value = UnitTestPreconditions.Jdk23OrEarlier, reason = "Scala does not work with Java 24 without warnings yet")
```

Comment:
For all the tests marked `Scala does not work with Java 24 without warnings yet`, does it make sense to use something like 

`ScalaCoverage.assumeIsCompatibleWithCurrentJdk(defaultScalaVersion)` 

so that when we _do_ update the scala version and it does become compatible with Java 24, we don't have to go and change the precondition for all these tests?

Reply:
I need to go back and adjust this and the JaCoCo tests -- essentially so we don't ever have a situation where we don't run the tests because we don't have any tested JDK version that's compatible with the Scala versions, I actually wanted to use preconditions here, since there's already a process to ensure we run a specific precondition at least once. But I need to add a precondition that uses the compatibility check for Scala/JaCoCo. I'll do that in a follow-up PR.